### PR TITLE
 Support for password recovery using claims other than the username

### PR DESCRIPTION
**Pull request by DinikaSen**
This PR adds the API client code required to support 'password recovery using claims other than the username' feature via the recovery endpoint.

Related to : https://github.com/wso2/identity-apps/pull/1245 
Resolves : https://github.com/wso2/product-is/issues/9267

### CODE REVIEW

**Comment by DinikaSen** [comment]
PR build passed : https://jenkins-support-all.wso2.com/jenkins/job/DEV_PR_BUILDER/959/

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointUtil.java [hunk 1]
```diff
@@ -692,4 +692,24 @@ public static String getBasePath(String tenantDomain, String context, boolean is
...
         return basePath;
     }
+
+    /**
+     * Get a query parameter value from a URL.
+     *
+     * @param url               URL
+     * @param queryParameter    Required query parameter name
+     * @return Query parameter value
+     * @throws URISyntaxException
```
Comment (ShanChathusanda93) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointUtil.java:702]: Shall we make this comment formatted like the other comments. Like having full stops at the EOL, having exception explanation comment. WDYT?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/api/PasswordRecoveryApiV1.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+    /**
+     * This API can be used to initiate recovering forgotten password/username.
+     *
+     * @param recoveryInitRequest Recovery initiating request. (required)
+     * @param tenantDomain        Tenant Domain which user belongs. Default &#x60;carbon.super&#x60; (optional)
+     * @param headers             If reCaptcha respond is found, embedded in request header. (optional)
+     * @param localVarPath        Endpoint path.
+     * @return Account recovery options response object.
+     * @throws ApiException if fails to make API call.
```
Comment (ShanChathusanda93) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/api/PasswordRecoveryApiV1.java:110]: Shall we start the comment with an uppercase letter. WDYT?

Comment (DinikaSen) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/api/PasswordRecoveryApiV1.java]: Addressed in https://github.com/wso2/carbon-identity-framework/pull/3171/commits/09b57785a1049994143d57f2e0620c3f88376e14

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointUtil.java (hunk 1 above)
Comment (DinikaSen) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointUtil.java]: Addressed in https://github.com/wso2/carbon-identity-framework/pull/3171/commits/09b57785a1049994143d57f2e0620c3f88376e14

**Review by ashensw — APPROVED** [review]

**Pull request merged**